rest/data/project_settings.go

```diff
@@ -142,15 +142,14 @@ func (sc *DBConnector) SaveProjectSettingsForSection(ctx context.Context, projec
 			}
 		}
 	case model.ProjectPageVariablesSection:
-		// Remove any variables that only exist in the original settings.
-		toDelete := []string{}
-		for key, _ := range before.Vars.Vars {
-			if _, ok := changes.Vars.Vars[key]; !ok {
-				toDelete = append(toDelete, key)
+		for key, value := range before.Vars.Vars {
```

Comment:
what would happen here in the case where the value was deleted?

Reply:
Yep! changes reflects the current state of the project, and if the variable is deleted from vars it should also be deleted from private vars. (Checking changes.Vars.Vars isn't quite as clear because the value has been made empty in order to redact the private value)